hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1453,7 +1453,11 @@ public static String deleteMetadataTablePartition(HoodieTableMetaClient dataMeta
    * @return The fileID
    */
   public static String getFileIDForFileGroup(MetadataPartitionType partitionType, int index) {
-    return String.format("%s%04d", partitionType.getFileIdPrefix(), index);
+    if (partitionType == MetadataPartitionType.FILES) {
+      return String.format("%s%04d-%d", partitionType.getFileIdPrefix(), index, 0);
+    } else {
```

Comment:
What is the purpos of extra suffix `-0` for `FILES` file group?

Reply:
Discuss offline, this is a fix for consistency of base file fileGroup id and the log files, because the bulk_insert create handle would append a suffix (starts from `-0`) for each file it creates.

Reply:
Should we fix all the partitions or just the `FILES` partition?

Reply:
yes, we should. guess I did push out a fix, but guess Sagar force pushed it. 
We have been dragging to land this patch for sometime. Lets go ahead if we get CI green. We will fix it in the next patch immediately. https://github.com/apache/hudi/pull/8900